pr_testing/run-pr-unittests.sh

```diff
@@ -0,0 +1,65 @@
+  echo '--------------------------------------'
+  mkdir -p $WORKSPACE/gpuUnitTests
+  let UT_TIMEOUT=7200+${CMSSW_PKG_COUNT}*20
+  UTESTS_CMD="USER_UNIT_TESTS=cuda timeout ${UT_TIMEOUT} scram b -v -k -j ${NCPU}  runtests "
+  echo "LD_LIBRARY_PATH: ${LD_LIBRARY_PATH}"
+  scram build echo_LD_LIBRARY_PATH || true
+  cms_major=$(echo ${CMSSW_IB} | cut -d_ -f2)
+  cms_minor=$(echo ${CMSSW_IB} | cut -d_ -f3)
+  cms_ver="$(echo 00${cms_major} | sed -E 's|^.*(..)$|\1|')$(echo 00${cms_minor} | sed -E 's|^.*(..)$|\1|')"
+  if [ $cms_ver -ge 1301 ] ; then
+    find $CMSSW_BASE/src -type d | grep -v '/__pycache__/*' | xargs chmod -w
+  fi
+  echo $UTESTS_CMD > $WORKSPACE/gpuUnitTests/log.txt
+  (eval $UTESTS_CMD && echo 'ALL_OK') > $WORKSPACE/gpuUnitTests/log.txt 2>&1 || true
+  echo 'END OF UNIT TESTS'
+  echo '--------------------------------------'
+  if [ $cms_ver -ge 1301 ] ; then
+    find $CMSSW_BASE/src -type d | xargs chmod +w
+  fi
+
+  TEST_ERRORS=$(grep -ai 'had errors\|recipe for target' $WORKSPACE/gpuUnitTests/log.txt | sed "s|'||g;s|.*recipe for target *||;s|.*unittests_|---> test |;s| failed$| timeout|" || true)
+  TEST_ERRORS=`grep -ai "had errors" $WORKSPACE/gpuUnitTests/log.txt` || true
+  GENERAL_ERRORS=`grep -a "ALL_OK" $WORKSPACE/gpuUnitTests/log.txt` || true
+
+  if [ "X$TEST_ERRORS" != "X" -o "X$GENERAL_ERRORS" = "X" ]; then
+    echo "Errors in the gpu unit tests"
+    echo 'UNIT_TEST_RESULTS;ERROR,Unit Tests,See Log,gpuUnitTests' >> ${RESULTS_DIR}/unittest.txt
+    ALL_OK=false
+    UNIT_TESTS_OK=false
+    $CMS_BOT_DIR/report-pull-request-results PARSE_UNIT_TESTS_FAIL -f $WORKSPACE/gpuUnitTests/log.txt --report-file ${RESULTS_DIR}/14-report.res ${REPORT_OPTS}
+    echo "UnitTests" > ${RESULTS_DIR}/14-failed.res
+  else
+    echo 'UNIT_TEST_RESULTS;OK,Unit Tests,See Log,gpuUnitTests' >> ${RESULTS_DIR}/unittest.txt
+  fi
+  echo "<html><head></head><body>" > $WORKSPACE/gpuUnitTests/success.html
+  cp $WORKSPACE/gpuUnitTests/success.html $WORKSPACE/gpuUnitTests/failed.html
+  UT_ERR=false
+  utlog="testing.log"
+  for t in $(find $WORKSPACE/$CMSSW_IB/tmp/${SCRAM_ARCH}/src -name ${utlog} -type f | sed "s|$WORKSPACE/$CMSSW_IB/tmp/${SCRAM_ARCH}/||;s|/${utlog}$||") ; do
+    mkdir -p $WORKSPACE/gpuUnitTests/${t}
+    mv $WORKSPACE/$CMSSW_IB/tmp/${SCRAM_ARCH}/${t}/${utlog} $WORKSPACE/gpuUnitTests/${t}/
+    if [ $(grep -a '^\-\-\-> test  *[^ ]*  *succeeded$' $WORKSPACE/gpuUnitTests/${t}/${utlog} | wc -l) -gt 0 ] ; then
+      echo "<a href='${t}/${utlog}'>${t}</a><br/>" >> $WORKSPACE/gpuUnitTests/success.html
+    else
+      echo "<a href='${t}/${utlog}'>${t}</a><br/>" >> $WORKSPACE/gpuUnitTests/failed.html
+      UT_ERR=true
+    fi
+  done
+  if ! $UT_ERR ; then echo "No unit test failed" >> $WORKSPACE/gpuUnitTests/failed.html ; fi
+  echo "</body></html>" >> $WORKSPACE/gpuUnitTests/success.html
+  echo "</body></html>" >> $WORKSPACE/gpuUnitTests/failed.html
+  prepare_upload_results
```

Comment:
we need to upload `gpuUnitTests` directory too. I think you need to update `prepare_upload_results` to upload this directory is exists